pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -555,6 +750,67 @@ public void uploadSegmentAsMultiPart(FormDataMultiPart multiPart,
     }
   }
 
+  @POST
+  @ManagedAsync
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.MULTIPART_FORM_DATA)
+  @Path("/segmentList")
+  @Authorize(targetType = TargetType.TABLE, paramName = "tableName", action = Actions.Cluster.UPLOAD_SEGMENT)
+  @Authenticate(AccessType.CREATE)
+  @ApiOperation(value = "Upload a segment", notes = "Upload a segment as binary")
```

Comment:
update the notes "Upload a batch of segments"

high level question: is this batch uploading API atomic? I see there is logic to clean up uploaded segments upon any failure in the middle, so it's atomic in terms of the uploading action, but I assume it's not atomic as to the ongoing queries, as I didn't see SegmentLineage or similar mechanism was used here. But anyway, might be helpful to comment the API about how things work upon failures.

Reply:
> update the notes "Upload a batch of segments"
> 
> high level question: is this batch uploading API atomic? I see there is logic to clean up uploaded segments upon any failure in the middle, so it's atomic in terms of the uploading action, but I assume it's not atomic as to the ongoing queries, as I didn't see SegmentLineage or similar mechanism was used here. But anyway, might be helpful to comment the API about how things work upon failures.

Good point. This is all or nothing API if I understand @rajagopr ?

Reply:
@klsince and @swaminathanmanish: As @Jackie-Jiang explained, the batch upload would work along with the segment start/end replace. The guarantee `all or nothing` is provided by the segment start/end replace protocol? I will call this out in the docs.